flake8_plugins/assert_expected_actual.py

```diff
@@ -0,0 +1,107 @@
+import ast
+import json
+from pathlib import Path
+import re
+
+
+MSG = 'PTA100 expected should come before actual'
+
+
+def is_literal(node):
+    return (
+        isinstance(node, ast.Constant) or
+        (
+            (
+                isinstance(node, ast.List) or
+                isinstance(node, ast.Set) or
+                isinstance(node, ast.Tuple)
+            ) and
+            all(is_literal(elt) for elt in node.elts)
+        ) or
+        (
+            isinstance(node, ast.Dict) and
+            all(is_literal(key) for key in node.keys) and
+            all(is_literal(value) for value in node.values)
+        )
+    )
+
+
+def classify_arg(node):
+    info = {'type': type(node).__name__, 'literal': is_literal(node)}
+    if info['type'] == 'Name':
+        for keyword in ['actual', 'expected']:
+            if re.search(keyword, node.id, re.IGNORECASE):
+                info[keyword] = True
+    return info
+
+
+def known_pattern(left, right):
+    return right.get('actual') or left.get('expected') or left.get('literal')
+
+
+def classify_call(node):
+    info = {'args': len(node.args)}
+    if info['args'] == 2:
```

Comment:
Are there many instances in code where assertEqual can take three arguments where the last is the message? If so, would it be easy to include that case here by relaxing the condition?

Reply:
according to the data from the PR description, there are 22 instances of `assertEqual` being used with three arguments; if you run the mentioned `flake8 --all-assert-equal-usages=/tmp/asserts` command, you should be able to grep in `/tmp/asserts` for something like `"args": 3` to find all of them